src/platform/cmx/boot_comp.c

```diff
@@ -0,0 +1,279 @@
+			    BOOT_CAPTBL_SELF_INITTCAP_BASE, 0, 1);
+	assert(!ret);
+
+	/*
+	 * boot component's mapped into SELF_PT,
+	 * switching to boot component's pgd
+	 */
+	cap_pt = (struct cap_pgtbl *)captbl_lkup(ct, BOOT_CAPTBL_SELF_PT_META);
+	if (!cap_pt || !CAP_TYPECHK(cap_pt, CAP_PGTBL)) assert(0);
+	/* PRY: modified to pass compilation */
+	//pgtbl = cap_pt->pgtbl;
+	assert(pgtbl);
+	pgtbl_update(pgtbl);
+
+	printk("\tCreating initial threads, tcaps, and rcv end-points in boot-component.\n");
+}
+/* PRY: the booing allocations have changed a lot. Not as usual, we do not allocate pages for CM3 */
+void
+kern_boot_comp(void)
+{
+	int ret = 0, nkmemptes;
+	struct captbl *ct;
+	unsigned int i;
+	u8_t *boot_comp_captbl;
+	void *thd_mem, *tcap_mem;
+	/* PRY:don't think that we need this */
+//	pgtbl_t pgtbl   = (pgtbl_t)chal_va2pa(&boot_comp_pgd), boot_vm_pgd;
+	u32_t hw_bitmap = 0xFFFFFFFF;
+	/* PRY:Allocate some memory for the booter component */
+	printk("Setting up the booter component.\n");
+
+	boot_comp_captbl = mem_boot_alloc(BOOT_CAPTBL_NPAGES);
+	assert(boot_comp_captbl);
+	ct               = captbl_create(boot_comp_captbl);
+	assert(ct);
+
+	/* Do not expand the captbl to use multiple pages - we know that it is enough for CMX. */
+//	for (i = PAGE_SIZE ; i < BOOT_CAPTBL_NPAGES*PAGE_SIZE ; i += PAGE_SIZE) {
+//		captbl_init(boot_comp_captbl + i, 1);
+//		ret = captbl_expand(ct, (i - PAGE_SIZE/2)/CAPTBL_LEAFSZ, captbl_maxdepth(), boot_comp_captbl + i);
+//		assert(!ret);
+//		captbl_init(boot_comp_captbl + PAGE_SIZE + PAGE_SIZE/2, 1);
+//		ret = captbl_expand(ct, i/CAPTBL_LEAFSZ, captbl_maxdepth(), boot_comp_captbl + i + PAGE_SIZE/2);
+//		assert(!ret);
+//	}
+
+	thd_mem  = mem_boot_alloc(1);
+	tcap_mem = mem_boot_alloc(1);
+	assert(thd_mem && tcap_mem);
+
+	if (captbl_activate_boot(ct, BOOT_CAPTBL_SELF_CT)) assert(0);
+	if (sret_activate(ct, BOOT_CAPTBL_SELF_CT, BOOT_CAPTBL_SRET)) assert(0);
+
+	hw_asndcap_init();
+	if (hw_activate(ct, BOOT_CAPTBL_SELF_CT, BOOT_CAPTBL_SELF_INITHW_BASE, hw_bitmap)) assert(0);
+
+	/*
+	 * separate pgd for boot component virtual memory
+	 */
+	/* PRY:No need to allocate pages for it. The capability table contains the page table.
+	 * The corasest granularity allowed by us is 128kB pages. Thus, we just add all these pages to the booter
+	 * component's page table  */
+	//boot_vm_pgd = (pgtbl_t)mem_boot_alloc(1);
+	//assert(boot_vm_pgd);
+	//memcpy((void *)boot_vm_pgd + KERNEL_PGD_REGION_OFFSET,  (void *)(&boot_comp_pgd) + KERNEL_PGD_REGION_OFFSET, KERNEL_PGD_REGION_SIZE);
+	//if (pgtbl_activate(ct, BOOT_CAPTBL_SELF_CT, BOOT_CAPTBL_SELF_PT, ,(pgtbl_t)chal_va2pa(boot_vm_pgd), 0)) assert(0);
+	/* Now the add address is hard-coded, add 4 big pages to the booter component. The booter component have access to all the memory
+	 * except for the kernel data */
+	if (pgtbl_activate(ct, BOOT_CAPTBL_SELF_CT, BOOT_CAPTBL_SELF_PT_META, COS_PGTBL_MPUMETA, 0x20020000, COS_PGTBL_SIZE_128K, COS_PGTBL_NUM_2)) assert(0);
```

Comment:
because the capability size have changed - the page table cap is now twice as large as before.